Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -14,16 +14,59 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string args, int expected)
-        {
+        private static void Test(string args, int expected)
+        {
             var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
-            Console.WriteLine($"{args} : {result}");
+            Console.WriteLine($"{args} : {result}");
         }
 
-        public static int FindDigit(string equation)
-        {
+        public static int FindDigit(string equation)
+        {         
             // Add your code here.
-            throw new NotImplementedException();
+            //Splitting The equation into three variables eg. 42*47=1?74 to A=42 B=47 and C=1?74
+            var A = equation.Split('*')[0];
+            var B = equation.Split('*')[1].Split('=')[0];
+            var C = equation.Split('=')[1];
+            var result = -1;
+            var ans = 0.0;
+            String originalValue = String.Empty, evaluatedValue;
+
+            //Checking which string variable contain "?" mark and storing the evaluated result in ans
+            if(A.Contains("?"))
+            {
+                ans = double.Parse(C)/double.Parse(B);
+                originalValue = A;
+            }
+            else if(B.Contains("?"))
+            {
+                ans = double.Parse(C)/double.Parse(A);

```

Comment:
What if A is 0?

Reply:
If either of A or B is Zero then C is definitely 0  so this condition applied before    any operation performed
if(C.Equals(0))
                return result;
 
where result is -1

Reply:
For that i can give a condition like
if(A.Equals(0) || B.Equals(0))
                return result;
where result is -1